src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UsePropertyInsteadOfCountMethodWhenAvailableTests.cs

```diff
@@ -646,5 +647,75 @@ public class SomeClass
                     .WithLocation(8, 23)
                     .WithArguments(nameof(IReadOnlyCollection<int>.Count)));
         }
+
+        [Fact, WorkItem(3724, "https://github.com/dotnet/roslyn-analyzers/issues/3724")]
+        public static async Task PropertyAccessParentIsNotAlwaysDirectlyTheInvocation()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class C
+{
+    public static bool IsChildPath(string parentPath, string childPath, IEnumerable<string> obj)
+    {
+        return (IsDirectorySeparator(childPath[parentPath.Length]) ||
+            IsDirectorySeparator(childPath[{|CA1829:parentPath.Count()|}])) ||
+            0.Equals((uint)obj.Count());
+    }
+
+    public static bool IsDirectorySeparator(char c) => false;
+}",
+                FixedCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class C
+{
+    public static bool IsChildPath(string parentPath, string childPath, IEnumerable<string> obj)
+    {
+        return (IsDirectorySeparator(childPath[parentPath.Length]) ||
+            IsDirectorySeparator(childPath[parentPath.Length])) ||
+            0.Equals((uint)obj.Count());
+    }
+
+    public static bool IsDirectorySeparator(char c) => false;
+}",
+            }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                TestCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Public Class C
+    Public Shared Function IsChildPath(ByVal parentPath As String, ByVal childPath As String, ByVal obj As IEnumerable(Of String)) As Boolean
+        Return (IsDirectorySeparator(childPath(parentPath.Length)) OrElse IsDirectorySeparator(childPath({|CA1829:parentPath.Count()|})) OrElse 0.Equals(CUInt(obj.Count()))
+    End Function
+
+    Public Shared Function IsDirectorySeparator(ByVal c As Char) As Boolean
+        Return False
+    End Function
+End Class
```

Comment:
I've a PR in roslyn for an analyzer that removes ByVal. I'm not aware if this test is run on the newer Roslyn version and the new analyzer flagged ByVal to be removed will that cause this test to fail due to reported diagnostics that isn't expected?

Consider removing ByVal if it can cause a problem in the future, or ignore this suggestion if it won't affect the unit test.

Reply:
Except if the rule is on by default and is a `CS` or `BC` rule it shouldn't impact here. But thank you for the input I will make the change.